daemon/algod/api/algod.oas2.json

```diff
@@ -2199,6 +2258,20 @@
         }
       }
     },
+    "SimulationResult": {
+      "description": "Result of a transaction group simulation",
+      "type": "object",
+      "properties": {
+        "failure-message": {
+          "description": "\\[fm\\] Failure message, if the transaction would have failed during a live broadcast.",
+          "type": "string"
+        },
+        "signature-failure-message": {
+          "description": "\\[sfm\\] Signature failure message, if the signature check would have failed.",
+          "type": "string"
+        }
+      }
```

Comment:
I completely agree, and that's where this endpoint is headed. There will need to be more details on each transaction and whether it passed or succeeded. This PR serves as the first step toward implementing that functionality, and focuses specifically on bootstrapping the basic structure of the simulate endpoint.

Providing details on which transaction failed (and at what PC) is functionality which will be significantly easier to implement with more advanced debugger hooks than we currently have – debugger hooks that I have already drafted and will be added in the next PR. I was hoping to hold off on implementing any form of trace information (including applydata/errors/etc) until those hooks are added.

The goal of these minimal fields is to easily check if the transaction group would have passed if broadcasted to the network as is. If both fields are empty, you're good to go. If one or both is not empty, you know to check the individual transaction details which will be added in future PRs.

That being said, signature verification is separate from transaction evaluation and it might be possible to include details in the `signature-failure-message` as to which transaction in the group had an incorrect signature. My plan was to add a `signature-valid` field to each transaction in the response, but I'm not opposed to including it here in the overarching signature error message.